src/Core/Banshee.WebBrowser/Banshee.WebSource/WebView.cs

```diff
@@ -26,64 +26,89 @@
 // THE SOFTWARE.
 
 using System;
-
-using Gtk;
-
-using Hyena;
-using Hyena.Downloader;
-
+using System.Runtime.InteropServices;
 using Banshee.Base;
-using Banshee.IO;
 using Banshee.WebBrowser;
+using Gdk;
+using GLib;
+using Hyena.Downloader;
+using EventArgs = System.EventArgs;
+using EventHandler = System.EventHandler;
+using Log = Hyena.Log;
 
 namespace Banshee.WebSource
 {
     public abstract class WebView : OssiferWebView
     {
-        protected string FixupJavascriptUrl { get; set; }
+        private const float ZoomStep = 0.05f;
         private string fixup_javascript;
         private bool fixup_javascript_fetched;
-
-        public bool IsReady { get; private set; }
-        public bool CanSearch { get; protected set; }
-
-        public event EventHandler Ready;
+        private double smooth_scroll_size;
 
         public WebView ()
         {
             CanSearch = false;
         }
 
-        const float ZOOM_STEP = 0.05f;
+        protected string FixupJavascriptUrl { get; set; }
+
+        public bool IsReady { get; private set; }
+        public bool CanSearch { get; protected set; }
+
+        public event EventHandler Ready;
 
         public void ZoomIn ()
         {
-            Zoom += ZOOM_STEP;
+            Zoom += ZoomStep;
         }
 
         public void ZoomOut ()
         {
-            Zoom -= ZOOM_STEP;
+            Zoom -= ZoomStep;
         }
 
-        protected override bool OnScrollEvent (Gdk.EventScroll scroll)
+        [DllImport ("libgdk-3-0.dll", CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr gdk_event_get_scroll_deltas (IntPtr eventHandle, out double deltaX,
+            out double deltaY);
+
+        protected override bool OnScrollEvent (EventScroll scroll)
         {
-            if ((scroll.State & Gdk.ModifierType.ControlMask) != 0) {
-                Zoom += (scroll.Direction == Gdk.ScrollDirection.Up) ? ZOOM_STEP : -ZOOM_STEP;
+            if ((scroll.State & ModifierType.ControlMask) != 0) {
+                switch (scroll.Direction) {
+                    case ScrollDirection.Up:
+                        ZoomIn ();
+                        break;
+                    case ScrollDirection.Down:
+                        ZoomOut ();
+                        break;
+                    case (ScrollDirection) 4: //GDK_SCROLL_SMOOTH
+                        double delta_x;
+                        double delta_y;
+                        gdk_event_get_scroll_deltas (scroll.Handle, out delta_x, out delta_y);
+                        smooth_scroll_size += delta_y;
+
+                        if (smooth_scroll_size < -1) {
+                            ZoomIn ();
+                            smooth_scroll_size += 1;
+                        }
+                        else if (smooth_scroll_size > 1) {
```

Comment:
Another one that's not mentioned in the style guidelines - everywhere in Banshee that I've read does:

```c#
if (condition1) {
} else if (condition2) {
} else {
}
```

We should stick with that, I'll add a mention in the guidelines for it. Just an FYI on this one, I use VS all day so I feel your pain on some of these, and yeah I do need to tidy up some of the stuff I've committed...

Reply:
Alright. Another change for the R# PR.

Reply:
added this one to the contribution guidelines #39